ios/Classes/RadarFlutterPlugin.m

```diff
@@ -583,7 +583,7 @@ - (void)getContext:(FlutterMethodCall *)call withResult:(FlutterResult)result {
     NSDictionary *argsDict = call.arguments;
 
     NSDictionary *locationDict = argsDict[@"location"];
-    if (locationDict) {
+    if (locationDict && [locationDict isKindOfClass:[NSDictionary class]]) {
         NSNumber *latitudeNumber = locationDict[@"latitude"];
```

Comment:
Should we do a `isKindOfClass` check on `locationDict[@"latitude"]` (and the following)?

Reply:
Hmm that is a great point, we actually do no enforcement of the types on the value in flutter (currently the values in dictionary are types dynamic, almost akin to any in js). We should definitely either enforce stronger types on the flutter level or the native level.

Reply:
Going to make this a breaking change as the language does not allow for overloads. The alternative is a structure like
```
static Future<Map?> searchGeofences(
      {dynamic near,
      int? radius,
      List? tags,
      Map<String, dynamic>? metadata,
      int? limit}) async {
    try {
      Map<String, dynamic>? nearMap;
      if (near is Map<String, dynamic>) {
        nearMap = near;
      } else if (near is RadarClass) {
        nearMap = near.toMap();
      } else if (near != null) {
        throw ArgumentError('near must be either a Map<String, dynamic> or a RadarClass');
      }

      return await _channel.invokeMethod('searchGeofences', <String, dynamic>{
        'near': nearMap,
        'radius': radius,
        'limit': limit,
        'tags': tags,
        'metadata': metadata
      });
    } on PlatformException catch (e) {
      print(e);
      return {'error': e.code};
    }
  }
  ```
 The part that really bothers me is that we are actually weakening the interface by typing the near field as dynamic, which still results in run-time issues instead of compile-time errors.